tests/integration/conftest.py

```diff
@@ -28,6 +39,87 @@ def fxt_open_subprocess(request: pytest.FixtureRequest) -> bool:
     return request.config.getoption("--open-subprocess")
 
 
+def find_recipe_folder(base_path: Path, folder_name: str) -> Path:
+    """
+    Find the folder with the given name within the specified base path.
+
+    Args:
+        base_path (Path): The base path to search within.
+        folder_name (str): The name of the folder to find.
+
+    Returns:
+        Path: The path to the folder.
+    """
+    for folder_path in base_path.rglob(folder_name):
+        if folder_path.is_dir():
+            return folder_path
+    msg = f"Folder {folder_name} not found in {base_path}."
+    raise FileNotFoundError(msg)
+
+
+def pytest_configure(config):
+    """Configure pytest options and set task, recipe, and recipe_ov lists.
+
+    Args:
+        config (pytest.Config): The pytest configuration object.
+
+    Returns:
+        None
+    """
+    task = config.getoption("--task")
+
+    # This assumes have OTX installed in environment.
+    otx_module = importlib.import_module("otx")
+    # Modify RECIPE_PATH based on the task
+    recipe_path = Path(inspect.getfile(otx_module)).parent / "recipe"
+    if task is not None:
+        recipe_path = find_recipe_folder(recipe_path, task.value.lower())
+        task_list = [task]
+    else:
+        task_list = [task_type for task_type in OTXTaskType if task_type != OTXTaskType.DETECTION_SEMI_SL]
+
+    # Update RECIPE_LIST
+    recipe_list = [str(p) for p in recipe_path.glob("**/*.yaml") if "_base_" not in p.parts]
+    recipe_ov_list = [str(p) for p in recipe_path.glob("**/openvino_model.yaml") if "_base_" not in p.parts]
+    recipe_list = set(recipe_list) - set(recipe_ov_list)
+
+    config.TASK_LIST = task_list
+    config.RECIPE_LIST = recipe_list
+    config.RECIPE_OV_LIST = recipe_ov_list
+
+
+def pytest_generate_tests(metafunc):
```

Comment:
Please use [pytest marker](https://docs.pytest.org/en/7.1.x/example/markers.html) to follow the common pytest practice rather than directly changing `pytest_generate_tests()`.

Reply:
Thanks! Done - https://github.com/openvinotoolkit/training_extensions/pull/2914/commits/c478f77fb340afc02f19fbe862f08b9da873c0b1
I removed pytest_generate_tests() in conftest and add to test function decorator as parameterize